tests/robustness/validate/patch_history.go

```diff
@@ -214,6 +214,9 @@ func putReturnTime(allOperations []porcupine.Operation, reports []report.ClientR
 					continue
 				}
 				kv := keyValue{Key: op.Put.Key, Value: op.Put.Value}
+				if count := persistedPutCount[kv]; count > 1 {
+					continue
+				}
```

Comment:
This is the only loop that cannot accept non-unique operations, however having it checked just in this place might seem odd. I think it would be clearer if we make it consistent and just not calculate returnTime for non-unique operations. 

Could you add this if block to all loops in `putReturnTime`? Maybe also rename `putReturnTime` to `uniquePutReturnTime`.